static/app/components/searchBar.tsx

```diff
@@ -30,6 +30,10 @@ function SearchBar({
 
   const [query, setQuery] = useState(queryProp ?? defaultQuery);
 
+  useEffect(() => {
+    setQuery(queryProp ?? '');
+  }, [queryProp]);
+
```

Comment:
I have a feeling this behaviour _might be_ uncovered in a lot of cases where other components use this core-ui component. IMO this might introduce bugs because of lack of coverage and how many things this touches, I'd probably add another prop such as `isControlled` then roll implementations in one by one as we check them. If you've already checked the implementations then ignore me though 😄 and I'll 👍 since we need it soon to fix a performance landing bug.

Reply:
Hmm what do you mean by lack of coverage? Initially I wanted to implement a controlled/uncontrolled value structure (see https://github.com/getsentry/sentry/pull/40682/commits/7c5ca55d56d0b6982e6e0a2cf67b36ef196b4db8), but then backtracked because it caused some tests to fail. Decided to do it this way instead because it's [how we implemented the component before I turned it into an FC](https://github.com/getsentry/sentry/pull/37955/files#diff-b40e0227969c456d0c63d05c98fd54e51c8fccec6eb12ffdacd162bab5138a28L37).

Reply:
oh I just suspected lacking some coverage it might be since usually people don't implicitly test typing/search in all specializations / usage of a core component like a searchbar, so it was a bit of a "maybe this is a problem" concern which generally leads me to be a bit more careful rolling out behaviour changes to a core component, your mileage may vary.

In terms of test failing, did you introduce the controlled/uncontrolled w/ the default matching the existing behaviour (which is uncontrolled if I'm reading this right)? That way you can opt things into changing the behaviour. It looks like you were setting up a memoized with a dependent on `controlledQuery`. I would just use an explicit `isControlled` boolean which basically indicates whether the external prop updates internal state or not, that shouldn't change the default behaviour if that is set to `isControlled: false` by default, no?

Reply:
Ohh okay I think I understand the source of the confusion now. So https://github.com/getsentry/sentry/pull/37955 broke an existing behavior in ` SearchBar`. Before it was merged, `SearchBar` can be either controlled or uncontrolled depending on whether you've passed in a `query` prop or not. The controlled case is handled in the [UNSAFE_componentWillReceiveProps method](https://github.com/getsentry/sentry/pull/37955/files#diff-b40e0227969c456d0c63d05c98fd54e51c8fccec6eb12ffdacd162bab5138a28L37). 

I misunderstood how that method worked and effectively removed it during the refactor in https://github.com/getsentry/sentry/pull/37955, so now `SearchBar` can't be controlled. If you pass in a `query` prop it will only be used as a default value. This PR fixes that: it makes `SearchBar` controllable again. The `useEffect` I have here is doing exactly the same thing as the  [UNSAFE_componentWillReceiveProps method](https://github.com/getsentry/sentry/pull/37955/files#diff-b40e0227969c456d0c63d05c98fd54e51c8fccec6eb12ffdacd162bab5138a28L37) before. So this PR doesn't introduce any new feature – which is why I don't think we need to worry about coverage here – it just fixes a bug that I caused.